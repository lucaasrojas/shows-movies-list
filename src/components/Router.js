import React from "react";
import { ShowsList, ShowDetail } from ".";
import { Route, Routes } from "react-router-dom";

const Router = () => {
    return (
        <Routes>
            <Route exact path="/" element={<ShowsList />} />
            <Route exact path="/:id" element={<ShowDetail />} />
        </Routes>
    );
};

export default Router;
