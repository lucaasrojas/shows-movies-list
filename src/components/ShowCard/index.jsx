import React from "react";
import { useNavigate } from "react-router-dom";
import CardStyles from "./ShowCardStyles";

const ShowCard = (props) => {
    const {
        show: { image, name, summary, id, rating }
    } = props;
    const navigate = useNavigate();
    const styles = CardStyles(props);
    return (
        <div data-testid={`${id}`} id={`${id}`} style={styles.wrapper} onClick={() => navigate(`/${id}`)}>
            {image ? (
                <img src={image?.medium} alt={`${name}`} style={styles.image} />
            ) : (
                <div style={styles.imagePlaceholder}></div>
            )}
            <div style={styles.textContainer}>
                <p style={styles.title}>{name}</p>
                {
                    props.detail && rating.average && (
                        <span style={styles.rating}>Rating: {rating.average}</span>
                    )
                }
                <p style={styles.summary} dangerouslySetInnerHTML={{ __html: summary }}></p>
            </div>
        </div>
    );
};

export default ShowCard;
