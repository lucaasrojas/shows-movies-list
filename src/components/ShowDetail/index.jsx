import React from "react";
import Api from "../../api";
import { ShowCard } from "../index";
import { useParams } from "react-router-dom";

const ShowDetail = () => {
    const [show, setShow] = React.useState();
    const queryParams = useParams();
    React.useEffect(() => {
        if (!show) {
            Api.getById({ id: queryParams.id }).then((res) => setShow(res));
        }
    }, [show, queryParams.id]);

    return show ? <ShowCard detail show={show} /> : null;
};

export default ShowDetail;
