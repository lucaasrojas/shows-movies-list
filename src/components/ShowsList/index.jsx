import React from "react";
import { ShowCard } from "../index";
import Api from "../../api";

const ShowsList = (props) => {
    const [showsList, setShowsList] = React.useState(props.showsList || []);

    React.useEffect(() => {
        if (!showsList.length) {
            Api.getAll().then((res) => {
                setShowsList(res);
            });
        }
    }, []);

    return showsList.length ? (
        <div>
            {showsList.map((el, i) => (
                <ShowCard data-testid={`${el.show.id}`} key={el.show.id} {...el} />
            ))}
        </div>
    ) : null;
};

export default ShowsList;
