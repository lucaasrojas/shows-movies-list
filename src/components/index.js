export { default as ShowsList } from "./ShowsList";
export { default as ShowDetail } from "./ShowDetail";
export { default as ShowCard } from "./ShowCard";
export { default as Router } from "./Router";
