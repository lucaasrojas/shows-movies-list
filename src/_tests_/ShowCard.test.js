
import 'regenerator-runtime/runtime'
import { render, queryByAttribute, fireEvent } from "@testing-library/react";
import React from "react";
import { ShowCard } from "../components";
import { BrowserRouter } from "react-router-dom";
const getById = queryByAttribute.bind(null, 'id')
const mockedUsedNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}));
it("With data should render the show card", async () => {
    const res = {
        show: {
            id: 123,
            image: {
                medium: "http://www.image.com"
            },
            name: "Show Name",
            summary: "<p>Brief description</p>"
        }
    };
    const wrapper = render(<BrowserRouter><ShowCard {...res} /></BrowserRouter>);
    const card = getById(wrapper.container, '123')
    expect(await wrapper.findByText(res.show.name)).toBeTruthy()
    expect(card).toBeTruthy();
    expect(wrapper).toMatchSnapshot()
});

it("On click should redirect to detail", async () => {
    const res = {
        show: {
            id: 123,
            image: {
                medium: "http://www.image.com"
            },
            name: "Show Name",
            summary: "<p>Brief description</p>"
        }
    };
    const wrapper = render(<BrowserRouter><ShowCard {...res} /></BrowserRouter>);
    const card = getById(wrapper.container, '123')
    fireEvent.click(card)

    expect(mockedUsedNavigate).toHaveBeenCalledWith('/123')

});