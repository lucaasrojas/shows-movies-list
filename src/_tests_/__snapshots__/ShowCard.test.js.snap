// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With data should render the show card 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="123"
        id="123"
        style="display: flex; padding: 1rem; width: 500px; border: 1px solid #bbb; border-radius: 10px; margin: 5px; cursor: pointer;"
      >
        <img
          alt="Show Name"
          src="http://www.image.com"
          style="height: 200px;"
        />
        <div
          style="display: flex; flex-direction: column; width: 100%; height: inherit; text-align: center; align-content: center; padding: 10px; overflow: hidden; white-space: nowrap;"
        >
          <p>
            Show Name
          </p>
          <p>
            <p>
              Brief description
            </p>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="123"
      id="123"
      style="display: flex; padding: 1rem; width: 500px; border: 1px solid #bbb; border-radius: 10px; margin: 5px; cursor: pointer;"
    >
      <img
        alt="Show Name"
        src="http://www.image.com"
        style="height: 200px;"
      />
      <div
        style="display: flex; flex-direction: column; width: 100%; height: inherit; text-align: center; align-content: center; padding: 10px; overflow: hidden; white-space: nowrap;"
      >
        <p>
          Show Name
        </p>
        <p>
          <p>
            Brief description
          </p>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
